pkg/pubsub/adapter/adapter.go

```diff
@@ -151,7 +165,17 @@ func (a *Adapter) receive(ctx context.Context, event cloudevents.Event, resp *cl
 func (a *Adapter) convert(ctx context.Context, m transport.Message, err error) (*cloudevents.Event, error) {
 	logger := logging.FromContext(ctx)
 	logger.Debug("Converting event from transport.")
+
 	if msg, ok := m.(*cepubsub.Message); ok {
+		if a.converters != nil && msg.Attributes != nil {
+			if val, ok := msg.Attributes["knative-gcp"]; ok {
+				logger.Infof("Converting Knative-GCP event: %q", val)
+				if c, ok := a.converters[val]; ok {
+					return c(ctx, m, err)
+				}
+			}
+		}
+
 		tx := cepubsub.TransportContextFrom(ctx)
 		// Make a new event and convert the message payload.
 		event := cloudevents.NewEvent()
```

Comment:
One question I had when writing this was is it important to pass along the topic and/or the subscription that delivered this? I could see a user encoding some logic in the name of the topic in some cases. But it does not make sense for like, a channel... What do you think?